src/pages/LogInWithShortLivedTokenPage.js

```diff
@@ -69,16 +65,20 @@ class LogInWithShortLivedTokenPage extends Component {
         }
 
         signInWithShortLivedToken(accountID, email, shortLivedToken, encryptedAuthToken);
-        this.setState({hasRun: true});
     }
 
     componentDidUpdate() {
-        if (this.state.hasRun || !this.props.betas) {
+        const email = lodashGet(this.props.route.params, 'email', '');
+        if (!this.props.betas || !this.props.session.authToken || email !== this.props.session.email) {
             return;
         }
 
         // exitTo is URI encoded because it could contain a variable number of slashes (i.e. "workspace/new" vs "workspace/<ID>/card")
-        const exitTo = decodeURIComponent(lodashGet(this.props.route.params, 'exitTo', ''));
+        let exitTo = decodeURIComponent(lodashGet(this.props.route.params, 'exitTo', ''));
+        if (exitTo === ROUTES.WORKSPACE_NEW) {
+            // New workspace creation is handled in AuthScreens, not in its own screen
+            exitTo = '';
```

Comment:
Just to clarify by setting this to `''` we are navigating from `/transition` and to `/` instead? 

Could we also just return here and wait for the logic in `AuthScreens` to run and redirect us to the workspace card page?